tracehandler/tracehandler.go

```diff
@@ -133,11 +132,11 @@ func (m *traceHandler) HandleTrace(bpfTrace *host.Trace) {
 
 	if !m.reporter.SupportsReportTraceEvent() {
 		// Fast path: if the trace is already known remotely, we just send a counter update.
-		postConvHash, traceKnown := m.bpfTraceCache.Get(bpfTrace.Hash)
```

Comment:
Doesn't this change our 128bit hashes we report to the backend completely?

Essentially, the hashing logic in `ConvertTrace` is no longer used, in favor of 

```
tmpHash := xxh3.Hash128(raw)
	trace.Hash = libpf.NewTraceHash(tmpHash.Hi, tmpHash.Lo)
```

which has different characteristics and cardinality.

Reply:
No, it does not affect the hash that is reported to the backend. If so, this proposed change would be a breaking change.

The hash, that is used to report back to the backend is untouched and happens in `ConvertTrace()` **after** the symbolization:

https://github.com/open-telemetry/opentelemetry-ebpf-profiler/blob/48471c0a9107eda4d5d0784819052e3ae9f1ae3f/processmanager/manager.go#L292

Reply:
But we're calling in line 139:

```
m.reporter.ReportCountForTrace(bpfTrace.Hash, 1, meta)
```

Reply:
good catch :+1: 

The hash returned by `ConvertTrace()` is stored for translation in `bpfTraceCache` and therefore this should not change. I will close the PR.